### PR TITLE
Added `--binary` argument for sending arbitrary bytes instead of text

### PR DESCRIPTION
**Pull request by jimni**
The main motivation for this feature was my need to send null bytes, which had been impossible in a null-terminated strings environment.

If you find this feature appropriate, I will be glad to update the PR according to your feedback.

This is my first experience with the elixir, sorry if the code sucks.

### CODE REVIEW

**Comment by savonarola** [comment]
Hello!
Thanks for the contribution.

Looks almost ready to be merged: there are some indentation issues. Could you fix them?

File: lib/smppsend.ex [hunk 1]
```diff
@@ -44,6 +44,7 @@ defmodule SMPPSend do
     udh_part_num: :integer,
 
     ucs2: :boolean,
+	binary: :boolean,
```
Comment (savonarola) [inline lib/smppsend.ex]: Indentation seems to be broken here

File: lib/smppsend.ex [hunk 2]
```diff
@@ -63,6 +64,7 @@ defmodule SMPPSend do
     udh_part_num: 1,
 
     ucs2: false,
+	binary: false,
```
Comment (savonarola) [inline lib/smppsend.ex]: And here

File: lib/smppsend.ex [hunk 3]
```diff
@@ -143,6 +146,23 @@ defmodule SMPPSend do
     end
   end
 
+  defp decode_hex_string(opts) do
+    if opts[:binary] do
+      case SMPPSend.OptionHelpers.decode_hex_string(opts, :short_message) do
+        {:ok, new_opts} ->
+          tlvs = opts[:tlvs]
+          {:ok, message_payload_id} = SMPPEX.Protocol.TlvFormat.id_by_name(:message_payload)
+        case SMPPSend.OptionHelpers.decode_hex_string(tlvs, message_payload_id) do
```
Comment (savonarola) [inline lib/smppsend.ex]: And here %)

**Comment by jimni** [comment]
Will do later today. Should I squash the commits?

**Comment by jimni** [comment]
Oh, ok. Thanks :)

**Pull request merged**